packages/core/src/types.ts

```diff
@@ -84,25 +84,40 @@ export interface ActionArgs<
   action: TAction;
 }
 
+export type InputFrom<T extends AnyActorLogic> = T extends ActorLogic<
+  infer _TEvent,
+  infer _TSnapshot,
+  infer _TInternalState,
+  infer _TPersisted,
+  infer _TSystem,
+  infer TInput
+>
+  ? TInput
+  : any; // TODO: never?
```

Comment:
hm, at first I was thinking about `undefined` but probably `unknown` would fit this even better (let's tinker with that in a separate PR though)

Reply:
actually, this type constraints the input already so this branch should just use `never` here (it's a dead branch anyway)